checkmk_extensions/agent/scheduler_runner.ps1

```diff
@@ -56,3 +56,87 @@ class Command {
         return [Command]::new($_executable, $_argumentsList)
     }
 }
+
+function StartSchedulerRunner {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$configFilePath
+    )
+
+    $configFileContent = [Config]::ParseConfigFile($configFilePath)
+
+    $command = [Command]::CreateCommand($configFileContent)
+    $parentDir = Split-Path $PSScriptRoot -Parent
+    $logPath = Join-Path $parentDir "log\robotmk\scheduler_runner.log"
+
+    # Create processInfo
+    $processInfo = New-Object System.Diagnostics.ProcessStartInfo
+    $processInfo.FileName = $command.Executable
+    $processInfo.Arguments = $command.ArgumentsList
+    $processInfo.RedirectStandardOutput = $true
+    $processInfo.RedirectStandardError = $true
+    $processInfo.UseShellExecute = $false
+    $processInfo.CreateNoWindow = $true
+    # $processInfo.WorkingDirectory -> Using this we can configure from where the Process will be started
+    $processInfo.WorkingDirectory = $PSScriptRoot
+
+    # Create process and assign processInfo to it
+    $process = New-Object System.Diagnostics.Process
+    $process.StartInfo = $processInfo
+
+
+    try {
+        $process.Start() | Out-Null
+
+        # Read stdout and stderr and handle them as needed
+        while ($true) {
+            $stdout = $process.StandardOutput.ReadLine()
+            if ($null -ne $stdout) {
+                # Handle stdout output here
+                WriteLog -Message $stdout -LogPath $logPath
+            }
+
+            $stderr = $process.StandardError.ReadLine()
+            if ($null -ne $stderr) {
+                # Handle stderr output here
+                WriteLog -Message $stderr -LogPath $logPath
+            }
+        }
+    }
+    catch {
+        # TODO: Handle errors
+        WriteLog -Message $_.Exception.Message -LogPath $logPath
+        throw $_.Exception.Message
+    }
+}
+
+function WriteLog {
```

Comment:
I think we will likely want to do more than logging (report the error to the Checkmk server in the form of a section), but this can stay as is for now. We can extend it in a follow-up step.